pythran/pythonic/types/numpy_expr.hpp

```diff
@@ -54,7 +54,7 @@ namespace pythonic {
                 typedef decltype(Op()(std::declval<typename std::remove_reference<Args>::type::dtype>()...)) dtype;
 
                 std::tuple<typename std::remove_reference<Args>::type...> args;
-                array<long, value> shape;
+                types::array<long, value> shape;
```

Comment:
I think it is better to have correct typing when it is possible as backend compiler will not it can't be negative and certainly do some tricky improvement (You spoke about signed/unsigned comparison and some kind of change like this)
I didn't do the modification here because it is a parameter from user input in some cases so it would lead to too many change in the code so I don't want to do it at the same time.